### PR TITLE
refactor(navy): use css sticky instead of js

### PR DESCRIPTION
**Pull request by SukkaW**
## Check List

**Please read and check followings before submitting a PR.**

- [ ] I want to publish my theme on Hexo official website.
    - [ ] I have read the [theme publishing doc](https://hexo.io/docs/themes#Publishing).
    - [ ] `name` is unique.
    - [ ] `link` URL is correct.
    - [ ] `preview` URL is correct.
    - [ ] `preview` URL web site is rendered correctly.
    - [ ] Add a screenshot to `source/themes/screenshots`.
    - [ ] Screenshot filename is same as value of `name`.
    - [ ] Screenshot size is `800 * 500`.
    - [ ] Screenshot file format is `png`.
- [ ] I want to publish my plugin on Hexo official website.
    - [ ] I have read the [plugin publishing doc](https://hexo.io/docs/plugins#Publishing).
    - [ ] `name` is unique.
    - [ ] `link` URL is correct.
- [x] Others (Update, fix, translation, etc...)
    - Languages:
    - [ ] `en` English
    - [ ] `ko` Korean
    - [ ] `pt-br` Brazilian Portuguese
    - [ ] `ru` Russian
    - [ ] `th` Thai
    - [ ] `zh-cn` simplified Chinese
    - [ ] `zh-tw` traditional Chinese

This PR removes `toc.js` by using css `position: sticky`.

Ref: https://developer.mozilla.org/en-US/docs/Web/CSS/position

Compatibility:

### CODE REVIEW

File: themes/navy/source/css/_partial/page.styl [hunk 1]
```diff
@@ -34,23 +34,20 @@ note-warn = hsl(0, 100%, 50%)
...
-    right: sidebar-width
+
 
 #article-toc-inner
   @extend .inner
   overflow-x: hidden
   overflow-y: auto
+  margin-top: 40px
+  margin-bottom: 40px
+  max-height: calc(100vh - 80px)
```
Comment (SukkaW) [inline themes/navy/source/css/_partial/page.styl:49]: Add `margin-(top|bottom)` for toc inner.

File: themes/navy/source/css/_partial/page.styl [hunk 2]
```diff
@@ -65,7 +62,6 @@ note-warn = hsl(0, 100%, 50%)
   text-decoration: none
   color: color-default
   display: block
-  margin-bottom: 40px
```
Comment (SukkaW) [inline themes/navy/source/css/_partial/page.styl:68]: Remove `margin-bottom` for "Back to Top" button, as the `margin-bottom` is already added for toc-inner.

File: themes/navy/source/css/_partial/sidebar.styl [hunk 3]
```diff
@@ -8,8 +8,11 @@
   @media mq-normal
     display: block
 
+  .inner
+    margin-top: 40px
+    margin-bottom: 40px
+
 .sidebar-title
-  margin-top: 40px
```
Comment (SukkaW) [inline themes/navy/source/css/_partial/sidebar.styl:12]: Remove `margin-top` as toc-inner & sidebar-inner already have `margin-(top|bottom)`

File: themes/navy/source/css/_partial/sidebar.styl [hunk 4]
```diff
@@ -8,8 +8,11 @@
   @media mq-normal
     display: block
 
+  .inner
+    margin-top: 40px
+    margin-bottom: 40px
```
Comment (SukkaW) [inline themes/navy/source/css/_partial/sidebar.styl:13]: Add `margin-(top|bottom)` for sidebar-inner.

**Comment by SukkaW** [comment]
Preview of the changes could be found on Netlify: https://deploy-preview-1452--hexo-site.netlify.app/

**Comment by jiangtj** [comment]
image

The scroll bar is displayed even if the length does not exceed the height of the window

**Comment by SukkaW** [comment]
> The scroll bar is displayed even if the length does not exceed the height of the window

@jiangtj  It turns out that it is caused by the `border` at the top of the `content` title.

File: themes/navy/source/css/_partial/sidebar.styl (hunk 3 above)
Comment (jiangtj) [reply themes/navy/source/css/_partial/sidebar.styl:12]: image

Is this the expected change?

Comment (SukkaW) [reply themes/navy/source/css/_partial/sidebar.styl:12]: @jiangtj Fixed.

File: themes/navy/source/css/_partial/sidebar.styl (hunk 4 above)
Comment (stevenjoezhang) [reply themes/navy/source/css/_partial/sidebar.styl:13]: `margin-top: 0` looks better
截屏2020-06-19 下午3 19 45

Comment (SukkaW) [reply themes/navy/source/css/_partial/sidebar.styl]: @stevenjoezhang 

Fixed & rebased.

**Pull request merged**